config.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"context"
+	"os"
+
+	"github.com/lema-ai/ippon/registry"
+	"github.com/pkg/errors"
+	"github.com/spf13/viper"
+)
+
+type Config struct {
+	ECR            *registry.ECR
+	ServicesConfig *ServicesConfig
+}
+
+type ServicesConfig struct {
+	GoServices     []GoServiceConfig     `mapstructure:"go_services"`
+	DockerServices []DockerServiceConfig `mapstructure:"docker_services"`
+}
+
+type DockerServiceConfig struct {
+	Tags       []string `mapstructure:"tags"`
+	Dockerfile string   `mapstructure:"dockerfile"`
+	// For multi-target build files.
+	// If nothing is passed here, will build a single target with the configured Name.
+	TargetsOrder []Target `mapstructure:"targets_order"`
+}
+
+type Target struct {
+	Name   string `mapstructure:"name"`
+	Target string `mapstructure:"target"`
+}
+
+func (this DockerServiceConfig) GetTags() []string {
+	if this.Tags != nil {
+		return this.Tags
+	}
+
+	return viper.GetStringSlice("tags")
+}
+
+type GoServiceConfig struct {
+	Name      string   `mapstructure:"name"`
+	Tags      []string `mapstructure:"tags"`
+	Main      string   `mapstructure:"main"`
+	BaseImage string   `mapstructure:"base_image"`
+}
+
+func (this GoServiceConfig) GetTags() []string {
+	if this.Tags != nil {
+		return this.Tags
+	}
+
+	return viper.GetStringSlice("tags")
+}
+
+func (this GoServiceConfig) GetBaseImage() string {
+	if this.BaseImage != "" {
+		return this.BaseImage
+	}
+
+	return viper.GetString("base_image")
+}
+
+func getConfig(registryName, path string) (*Config, error) {
+	f, err := os.Open(path)
```

Comment:
Why do we do all of this block instead of just calling ` viper.ReadInConfig()`?

Reply:
Because that way we can have multiple config files - useful for separating different builds for regular/llm